integration-cli/docker_cli_push_test.go

```diff
@@ -577,4 +577,6 @@ func (s *DockerRegistryAuthTokenSuite) TestPushMisconfiguredTokenServiceResponse
 	dockerCmd(c, "tag", "busybox", repoName)
 	out, _, err := dockerCmdWithError("push", repoName)
 	c.Assert(err, check.NotNil, check.Commentf(out))
+	split := strings.Split(out, "\n")
+	c.Assert(split[len(split)-2], check.Equals, "unauthorized: authentication required")
```

Comment:
@aaronlehmann funny enough, now a bad formatted error won't result always in an unexpected http error but with the statuscode it translates to the correct error message in the client, is it the expected behavior?